tests/conftest.py

```diff
@@ -96,17 +96,21 @@ def _init_state(n):
 
 @pytest.fixture(params=analytic_devices + hw_devices)
 def device(request, shots):
-    """Fixture to initialize and return a PennyLane device"""
+    """Fixture to initialize and return a PennyLane device with shots"""
     device = request.param
 
-    if device in analytic_devices:
-
-        def _device(n):
+    def _device(n):
             return device(wires=n, shots=shots)
 
-    elif device in hw_devices:
+    return _device
+
 
-        def _device(n):
-            return device(wires=n, shots=shots)
+@pytest.fixture(params=analytic_devices + hw_devices)
+def device_analytic(request):
+    """Fixture to initialize and return a PennyLane device without shots"""
+    device = request.param
+
+    def _device(n):
+            return device(wires=n, shots=None)

```

Comment:
```suggestion
            return device(wires=n)

```
I guess we shouldn't even be using the shots= kwarg here at all right?